src/frontends/common/src/plugin_loader.hpp

```diff
@@ -15,6 +15,23 @@ static const char PathSeparator[] = ":";
 namespace ov {
 namespace frontend {
 
+/// \brief Internal data structure holding by each frontend. Includes library handle and extensions.
+class FrontEndSharedData {
+    std::shared_ptr<void> m_so;
+    std::vector<std::shared_ptr<ov::Extension>> m_loaded_extensions = {};
+
+public:
+    explicit FrontEndSharedData(const std::shared_ptr<void>& so) : m_so(so) {}
+    std::vector<std::shared_ptr<ov::Extension>>& extensions() {
+        return m_loaded_extensions;
+    }
+};
+
+inline void add_extension_to_shared_data(std::shared_ptr<void>& obj, const std::shared_ptr<ov::Extension>& ext) {
+    auto obj_data = std::static_pointer_cast<FrontEndSharedData>(obj);
```

Comment:
Maybe we can change a type of `fe_obj->m_shared_objec` to `FrontEndSharedData` in order to avoid static casts.
Or just add an additional vector of shared_pointers to void into common Frontend class (as was done for `ir::FrontEnd`, it allows to avoid additional structures.

Reply:
Introduced chaning type of `m_shared_object` to avoid casting.
Storing the vector of extensions in Frontend seems to be worse solution because extenions should be keep not only util Frontend is alive but also until Model is alive so the current solution at least for me is more readable